File: Data Structures & Algorithms/two-integer-sum/submission-6.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> cnt;
        for(int i = 0; i < nums.size(); i++){
            int need = target - nums[i];
            if(cnt.count(need)) return {cnt[need], i};
            cnt[nums[i]] = i;
        }
        return {};
    }
};

File: Data Structures & Algorithms/duplicate-integer/submission-6.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        cnt = {}
        for x in nums:
            if x in cnt:
                cnt[x] += 1
            else:
                cnt[x] = 1
        for val, freq in cnt.items():
            if freq > 1:
                return True
        return False 